koongo-com/magento2-data-feed-manager
# Remove manual steps to install static assets (fonts), and automate th…

…e install using magento/magento-composer-installer

This PR
- removes the manual installation steps and script to copy font files in your Magento installation
- introduces the `magento/magento-composer-installer` dependency which can automate the install of font files in your Magento installation

This is a little-known feature in Magento, but you can see that Magento's `magento/magento2-base` module uses the same feature to copy files from it, to your Magento install.
The difference is that `magento/magento2-base` is of `type` `magento2-component`, while this module is of `type` `magento2-module`. The `type` determines the location where files are copied to as described in [the documentation](https://github.com/magento/magento-composer-installer/#supported-components). So I needed to prepend the path with `../../` so it would install the fonts in `{root-magento-project}/lib` and not `{root-magento-project}/app/code/lib/`

To test:
1. Add the following to the `repositories` section of the  `composer.json` file of a Magento installation:
```json
        {
            "type": "vcs",
            "url": "https://github.com/baldwin-agency/koongo-magento2-data-feed-manager"
        }
```
2. Verify that `lib/web/fonts/` does **not** has a directory `Nostress_Koongo` with the fonts from this module
3. Execute:
```
composer require koongo-com/magento2-data-feed-manager:dev-automate-install-of-custom-fonts
```
4. Verify that `lib/web/fonts/` **does** have a directory `Nostress_Koongo` with the fonts from this module and that in the Magento backoffice the font is loaded correctly
5. Execute:
```
composer remove koongo-com/magento2-data-feed-manager
```
6. Verify that `lib/web/fonts/` does **not** has a directory `Nostress_Koongo` with the fonts from this module
